pages/login.js

```diff
@@ -22,6 +24,10 @@ export default function Login() {
     }
   };
 
+  useEffect(() => {
+    setShowExtensionLogin(!!window.nostr);
+  })
```

Comment:
Unless there's something i'm missing?

Reply:
I'm fine either way, just could make it less verbose

Reply:
I initially tried to put it on line 98 but got a `window is undefined` error. Agreed it doesn't need to run on every render, but seems like the component needs to finish loading/initializing first?

Reply:
hmm that might have something to do with react strict mode. Basically, in development, the site gets rendered twice, once on the server and once on the client (this is to check for inconsistencies/give warnings or something idk really). So the first render would give that error since window is undefined on the server.

But anyway, you can just merge as is, it's really not a problem to do it how you did it.

If you want it to only run once (on first render) you add an empty dependency array like
```
  useEffect(() => {
    setShowExtensionLogin(!!window.nostr);
  }, [])
```

I think its fine to run on every render though, in case they enable/disable the extension while on the page or something.

Reply:
!!window?.nostr might work

Reply:
I'm happy to leave it as-is then, catching an enable/disable of the extension seems like a good argument.